src/io/read/mod.rs

```diff
@@ -261,6 +264,157 @@ extension_trait! {
         {
             ReadExactFuture { reader: self, buf }
         }
+
+        #[doc = r#"
+            Creates an adaptor which will read at most `limit` bytes from it.
+
+            This function returns a new instance of `Read` which will read at most
+            `limit` bytes, after which it will always return EOF ([`Ok(0)`]). Any
+            read errors will not count towards the number of bytes read and future
+            calls to [`read()`] may succeed.
+
+            # Examples
+
+            [`File`]s implement `Read`:
+
+            [`File`]: ../fs/struct.File.html
+            [`Ok(0)`]: ../../std/result/enum.Result.html#variant.Ok
+            [`read()`]: tymethod.read
+
+            ```no_run
+            use async_std::io::prelude::*;
+            use async_std::fs::File;
+
+            fn main() -> std::io::Result<()> {
```

Comment:
Generally we've been hiding the `fn main` and closing blocks from the examples. Could you maybe go over them and make sure those lines are prefixed with `#`?

Reply:
fixed